File: web-ui/src/app/_models/food-entry.ts
export default class FoodEntry {
  id: number;
  name: string;
  unit: string;
  quantity: number | null;
  unitCalorieValue: number | null;
  totalCalories: number | null;
  foodTakenAt: string | null;
  userId: string | null;
  userFullName?: string;
}

File: web-api/CalorieTracker.Core/Models/FoodEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CalorieTracker.Core.Models
{
    public class FoodEntry : ModelBase
    {

        [Required(ErrorMessage = "Food name is a required field.")]
        public string? Name { get; set; }

        public string? Unit { get; set; }
        public decimal? Quantity { get; set; }
        public decimal? UnitCalorieValue { get; set; }
        [Required(ErrorMessage = "Calorie value is a required field.")]
        public decimal? TotalCalories { get; set; }

        public DateTime? FoodTakenAt { get; set; }


        [ForeignKey(nameof(User))]
        public string? UserId { get; set; }

    }
}
